File: fsharp/log-levels/LogLevels.fs
module LogLevels

let message (logLine: string) : string =
    logLine.Split(':') |> Array.last |> (fun s -> s.Trim())

let logLevel (logLine: string) : string =
    logLine.Split(':')
    |> Array.head
    |> fun s -> s.Replace("[", "")
    |> fun s -> s.Replace("]", "")
    |> fun s -> s.ToLowerInvariant()

let reformat (logLine: string) : string =
    sprintf "%s (%s)" (message logLine) (logLevel logLine)

File: csharp/log-levels/LogLevels.cs
static class LogLine
{
    public static string Message(string logLine) =>
        logLine.Split(':')[1].Trim();

    public static string LogLevel(string logLine) =>
        logLine.Split(':')[0].Trim('[', ']').ToLowerInvariant();

    public static string Reformat(string logLine) =>
        $"{Message(logLine)} ({LogLevel(logLine)})";
}
